toolchain/check/import_cpp.cpp

```diff
@@ -197,6 +199,74 @@ static auto ClangLookup(Context& context, SemIR::LocId loc_id,
   return lookup;
 }
 
+// Creates an integer type of the given size.
+static auto MakeIntType(Context& context, IntId size_id) -> TypeExpr {
+  // TODO: Fill in a location for the type once available.
+  auto type_inst_id = MakeIntTypeLiteral(context, Parse::NodeId::None,
+                                         SemIR::IntKind::Signed, size_id);
+  return ExprAsType(context, Parse::NodeId::None, type_inst_id);
+}
+
+// Maps a C++ type to a Carbon type. Currently only 32-bit `int` is supported.
+// TODO: Support more types.
+static auto MapType(Context& context, clang::QualType type) -> TypeExpr {
+  const auto* builtin_type = dyn_cast<clang::BuiltinType>(type);
+  if (builtin_type && builtin_type->getKind() == clang::BuiltinType::Int &&
+      context.ast_context().getTypeSize(type) == 32) {
+    return MakeIntType(context, context.ints().Add(32));
+  }
+  return {.inst_id = SemIR::ErrorInst::SingletonInstId,
+          .type_id = SemIR::ErrorInst::SingletonTypeId};
+}
+
+// Returns a block id for the explicit parameters of the given function
+// declaration. If the function declaration has no parameters, it returns
+// `SemIR::InstBlockId::Empty`. In the case of an unsupported parameter type, it
+// returns `SemIR::InstBlockId::None`.
+static auto MakeParamPatternsBlockId(Context& context, SemIR::LocId loc_id,
+                                     const clang::FunctionDecl* clang_decl)
+    -> SemIR::InstBlockId {
+  if (clang_decl->parameters().empty()) {
+    return SemIR::InstBlockId::Empty;
+  }
+  SemIR::CallParamIndex next_index(0);
+  llvm::SmallVector<SemIR::InstId> params;
+  params.reserve(clang_decl->parameters().size());
+  for (clang::ParmVarDecl* param : clang_decl->parameters()) {
```

Comment:
BTW, can this be const, or is there a subtle potential mutation here?

```suggestion
  for (const clang::ParmVarDecl* param : clang_decl->parameters()) {
```

Reply:
It can be const. Thanks.